File: DSA/DSA Using Go/Insertion_sort.go
package main

import "fmt"

func InsertionSort(slice []int) {
	n := len(slice)
	for i := 1; i < n; i++ {
		cvalue := slice[i]
		position := i
		for position > 0 && slice[position-1] > cvalue {
			slice[position] = slice[position-1]
			position = position - 1
		}
		slice[position] = cvalue
	}
}

func main() {
	var slice = []int{3, 5, 8, 9, 6, 2}
	fmt.Println("Slice before sorting:", slice)
	InsertionSort(slice)
	fmt.Println("Slice after sorting:", slice)
}

File: DSA/DSA Using Python/Insertion_sort.py
def insertionsort(list):
    n = len(list)
    for i in range(1,n):
        cvalue = list[i]
        position = i
        while position > 0 and list[position-1] > cvalue:
            list[position] = list[position-1]
            position = position - 1
        list[position] = cvalue


list = [3,5,8,9,6,2]
print('Original Array:',list)
insertionsort(list)
print('Sorted Array:',list)